diff --git a/dlls/http.sys/http.c b/dlls/http.sys/http.c
--- a/dlls/http.sys/http.c
+++ b/dlls/http.sys/http.c
@@ -385,7 +385,7 @@ static BOOL host_matches(const struct url *url, const char *conn_host)
     if (!url->url)
         return FALSE;
 
-    if (url->url[7] == '+')
+    if (url->url[7] == '+' || url->url[7] == '*')
     {
         const char *queue_port = strchr(url->url + 7, ':');
         host_len = strchr(queue_port, '/') - queue_port - 1;
